src/vsr.zig

```diff
@@ -1076,6 +1062,8 @@ pub fn quorums(replica_count: u8) struct {
     assert(quorum_view_change >= @divFloor(replica_count, 2) + 1);
     assert(quorum_view_change + quorum_replication > replica_count);
 
+    // We need to have enough nacks to guarantee that `quorum_replication` was not reached,
+    // because if the replication quorum was reached, then it may have been committed.
```

Comment:
There might be ops preceding it in the pipeline which do not have a replication quorum of prepare-oks.
Or alternately, it the prepare oks might have been lost, and the op is "logically" committed (i.e. it will eventually be committed), but no replica has physically executed it in that state machine yet.

Reply:
Wrong: it might be truncated if some preceding entry is not committed.